hikari/api/entity_factory.py

```diff
@@ -103,7 +103,12 @@ class GatewayGuildDefinition:
 
 
 class EntityFactory(abc.ABC):
-    """Interface for components that serialize and deserialize JSON payloads."""
+    """Interface for components that serialize and deserialize JSON payloads.
+
+    !!! note
+        All functions in this class may raise `hikari.errors.UnrecognisedEntityError`
```

Comment:
Would they though? doesn't this only really apply to a few methods where this behaviour could be explicitly documented for them

Reply:
Since this is the API, we don't know how each implementation will want to raise it or not.

Reply:
depending on how "unrecognised entity" is interpreted this could be suggests that implementations should suppress all encountered errors and re-raise a UnrecognisedEntityError for all deserialization

Reply:
if it isn't meant to just cover cases of methods which dynamically map payloads to different types then the docs should show this but if it is then there isn't any reason why it can't just be documented on those methods